src/Scanner.js

```diff
@@ -194,9 +195,14 @@ class DefaultScanner {
             }
         }
 
+        /**
+         * Max number of nodes we want to traverse upwards, critical to avoid enclosing large portions of the DOM
+         * @type {number}
+         */
+        let traversalLayerCount = 0
         let element = input
         // traverse the DOM to search for related inputs
-        while (element.parentElement && element.parentElement !== document.documentElement) {
+        while (traversalLayerCount <= 5 && element.parentElement && element.parentElement !== document.documentElement) {
```

Comment:
This is the last important bit. We limit the upward traversal when searching for a form container. This decreases the chance of false-matching the `body` as a form container. For example, in Asana, with a specific sequence of clicks the algo could have matched the whole `body`, thus running extremely expensive checks on the whole DOM. With this change the algo stops at the rows of tasks, as expected. This ensures we hit the max form limit much quicker and have more tightly-scoped observers.